### ISSUE TITLE
Wizard Step: Create New Stream

### ISSUE THREAD

**Issue by grotlue** [Graylog2/graylog2-server#20569]

**Comment by grotlue** [comment]
@tellistone If the user chooses to "Create a new Pipeline for this Stream" what exactly is the content of the pipeline? Is it just an "empty" one with a name?

**Comment by grotlue** [comment]
APIs:
- http://localhost:8080/api/api-browser/global/index.html#!/Streams/create_post_1 to create a new stream

**Comment by tellistone** [comment]
> @tellistone If the user chooses to "Create a new Pipeline for this Stream" what exactly is the content of the pipeline? Is it just an "empty" one with a name?

Yes - empty pipeline with name that corresponds to the new stream, attached to the new stream

**Comment by grotlue** [comment]
Added backend label as we need to routes the messages to the new stream

**Comment by grotlue** [comment]
@tellistone Do I understand correctly that we want to ask the user if they want to create another custom stream?

**Comment by tellistone** [comment]
Yeah, their options are

1. no routing, same as today
2. pick an existing stream, we create routing to it
3. create a new stream, optionally attach an existing pipeline to it,  create routing to it

**Comment by tellistone** [comment]
In ui design, we should push the  "Create a New Index Set" action as highly recommended.

**Comment by grotlue** [comment]
For the contextual warning in the mockup there is no API. At least I couldn't find one that returns a list of streams connected to a specific index set. @patrickmann correct me if I'm wrong. I'd suggest to continue with this issue without that and if we want to have it, create a seperate one.

**Comment by patrickmann** [comment]
The stream APIs do include information about index sets. To find all streams for a specific index set, you currently have to call `GET /streams` and then filter by the desired index set.

The BE already has a method `indexSetIdsByIds` to return all stream IDs for a given set of index IDs, but it is not exposed as an API.

**Comment by grotlue** [comment]
I think it would be a better solution if we can either can integrate the streams in the index sets call or add a new call to get all streams for one index set.
I don't think it makes a lot of sense to get the whole stream list only for that use case.
